### PR TITLE
added log level for filelock

### PR DESCRIPTION
**Pull request by mirkobronzi**

### CODE REVIEW

**Comment by bouthilx** [comment]
What about setting the logging level for filelock inside `orion.core.io.database.pickleddb` instead?

**Comment by mirkobronzi** [comment]
> What about setting the logging level for filelock inside `orion.core.io.database.pickleddb` instead?

Yes, I think it makes sense. Just modified the code.

File: src/orion/core/io/database/__init__.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
    - :class:`orion.core.io.database.mongodb.MongoDB`
 
 """
+import logging
 from abc import abstractmethod, abstractproperty
```
Comment (bouthilx) [inline src/orion/core/io/database/__init__.py]: You should switch those lines to make flake8 happy.

File: src/orion/core/io/database/__init__.py [hunk 2]
```diff
@@ -264,3 +265,7 @@ class Database(AbstractDB, metaclass=SingletonFactory):
     """
 
     pass
+
+
+# set per-module log level
+logging.getLogger('filelock').setLevel('ERROR')
```
Comment (bouthilx) [inline src/orion/core/io/database/__init__.py]: Wait, ERROR? Isn't that always going to log? Maybe we should add a unit-test for this logging. :)

Comment (mirkobronzi) [reply src/orion/core/io/database/__init__.py]: it's the opposite actually..

given we cannot change the level in the module `filelock`, which is info, we change the required level for logging to error.

so, now, when the code runs on the line `logger.info`, info < error, and it does not print.

Comment (bouthilx) [reply src/orion/core/io/database/__init__.py]: Oh yes, my bad! I got confused. :sweat_smile:

File: tests/unittests/core/test_pickleddb.py [hunk 3]
```diff
@@ -263,6 +263,26 @@ def test_read_and_write_no_match(self, orion_db):
 
         assert loaded_config is None
 
+    def test_logging_when_getting_file_lock(self, caplog, orion_db):
+        """When logging.level is ERROR, there should be no logging."""
+        caplog.set_level(logging.INFO)
+        caplog.clear()
+        """Any operation will trigger the lock."""
```
Comment (bouthilx) [inline tests/unittests/core/test_pickleddb.py]: This should be a comment instead of a string.

File: tests/unittests/core/test_pickleddb.py [hunk 4]
```diff
@@ -263,6 +263,26 @@ def test_read_and_write_no_match(self, orion_db):
...
+        orion_db.read(
+            'experiments',
+            {'name': 'supernaedo2', 'metadata.user': 'dendi'})
+
+        assert 'acquired on orion_db.pkl.lock' in caplog.text
+
+        caplog.set_level(logging.ERROR)
+        caplog.clear()
+        """Any operation will trigger the lock."""
```
Comment (bouthilx) [inline tests/unittests/core/test_pickleddb.py]: Idem

File: tests/unittests/core/test_pickleddb.py (hunk 3 above)
Comment (mirkobronzi) [reply tests/unittests/core/test_pickleddb.py]: done :)

File: tests/unittests/core/test_pickleddb.py (hunk 4 above)
Comment (mirkobronzi) [reply tests/unittests/core/test_pickleddb.py]: done :)

**Comment by bouthilx** [comment]
Now needs rebase

**Pull request merged**